design/outlier-detection-design.md

```diff
@@ -0,0 +1,262 @@
+    // It can be disabled by setting the value to 0.
+    // Defaults to 5.
+    // +optional
+    Consecutive5xxErrors *uint32 `json:"consecutive5xxErrors,omitempty"`
+    
+    // Interval is the interval at which host status is evaluated.
+    // Defaults to 10s.
+    // +optional
+    // +kubebuilder:validation:Pattern=`^(((\d*(\.\d*)?h)|(\d*(\.\d*)?m)|(\d*(\.\d*)?s)|(\d*(\.\d*)?ms))+)$`
+    Interval *string `json:"interval,omitempty"`
+    
+    // BaseEjectionTime is the base time that a host is ejected for.
+    // A host will remain ejected for a period of time equal to the
+    // product of the ejection base duration and the number of times the host has been ejected.
+    // Defaults to 30s.
+    // +optional
+    // +kubebuilder:validation:Pattern=`^(((\d*(\.\d*)?h)|(\d*(\.\d*)?m)|(\d*(\.\d*)?s)|(\d*(\.\d*)?ms))+)$`
+    BaseEjectionTime *string `json:"baseEjectionTime,omitempty"`
+    
+    // MaxEjectionTime is the maximum time a host will be ejected for.
+    // After this amount of time, a host will be returned to normal operation.
+    // If not specified, the default value (300s) or BaseEjectionTime value is applied, whatever is larger.
+    // Defaults to 300s.
+    // +optional
+    // +kubebuilder:validation:Pattern=`^(((\d*(\.\d*)?h)|(\d*(\.\d*)?m)|(\d*(\.\d*)?s)|(\d*(\.\d*)?ms))+)$`
+    MaxEjectionTime *string `json:"maxEjectionTime,omitempty"`
+    
+    // SplitExternalLocalOriginErrors defines whether to split the local origin errors from the external origin errors.
+    // Defaults to false.
+    // +optional
+    // +kubebuilder:default=false
+    SplitExternalLocalOriginErrors bool `json:"splitExternalLocalOriginErrors"`
+    
+    // ConsecutiveLocalOriginFailure defines the number of consecutive local origin failures before a consecutive local origin ejection occurs.
+    // Parameters take effect only when SplitExternalLocalOriginErrors is true.
+    // Defaults to 5.
+    ConsecutiveLocalOriginFailure *uint32 `json:"consecutiveLocalOriginFailure,omitempty"`
+    
+    // MaxEjectionPercent is the max percentage of hosts in the load balancing pool for the upstream service.
+    // But will eject at least one host regardless of the value here.
+    // Defaults to 10%.
+    // +optional
+    // +kubebuilder:validation:Maximum=100
+    MaxEjectionPercent *uint32 `json:"maxEjectionPercent,omitempty"`
+    
+    // MinHealthPercent is the minimum percentage of hosts in the load balancing pool for the upstream service.
+    // If the number of hosts in the load balancing pool is less than the value set here, the outlier detection will not be performed.
+    // https://www.envoyproxy.io/docs/envoy/v1.26.2/intro/arch_overview/upstream/load_balancing/panic_threshold#arch-overview-load-balancing-panic-threshold
+    // Defaults to 0%.
+    // +optional
+    // +kubebuilder:validation:Maximum=100
+    // +kubebuilder:default=0
+    MinHealthPercent *uint32 `json:"minHealthPercent,omitempty"`
```

Comment:
I don't know if we need to offer configurability for panic threshold in this case since we have the above "max ejection %" config and we are not utilizing the panic threshold ability to "fail" traffic when in panic mode, is there a specific reason you have for adding this?

Reply:
Here is a reference to some of the design concepts of [istio](https://istio.io/latest/docs/reference/config/networking/destination-rule/#OutlierDetection). After careful consideration, I think we can remove this field and keep the default value of contour.Thanks  for your comment